doc/specs/vulkan/chapters/VK_KHR_swapchain/wsi.txt

```diff
@@ -436,11 +438,15 @@ endif::VK_KHX_device_group[]
 The pname:surface must: not be destroyed until after the swapchain is
 destroyed.
 
-If pname:oldSwapchain is not dlink:VK_NULL_HANDLE then pname:surface must:
-be associated with pname:oldSwapchain.
-Otherwise, the native window referred to by pname:surface must: not already
-be associated with another swapchain, and must: not be already be associated
-with a non-Vulkan graphics API surface.
+If pname:oldSwapchain is dlink:VK_NULL_HANDLE, the swapchain cannot: be
+created if the native window referred to by pname:surface is already
+associated with a swapchain.
+If pname:oldSwapchain is a valid swapchain, the swapchain cannot: be
```

Comment:
I think this condition is redundant with the VU statement you've added (with my suggested modifications). Right?

Reply:
Wrong.
This describes when `VK_ERROR_NATIVE_WINDOW_IN_USE_KHR` should be returned. It gets bit complicated to catch all corner cases:

1. `oldSwapchain == VK_NULL_HANDLE` and `surface` is not associated with any swapchain -> `VK_SUCCESS`
2. `oldSwapchain == VK_NULL_HANDLE` and `surface` associated with any (retired or non-retired) swapchain -> `VK_ERROR_NATIVE_WINDOW_IN_USE_KHR`
3. `oldSwapchain == valid` and `surface` is associated only with `oldSwapchain` -> `VK_SUCCESS`
4. `oldSwapchain == valid` and `surface` is associated only with `oldSwapchain` and other retired swapchains -> `VK_SUCCESS`
5. `oldSwapchain == valid` and `surface` is associated with non-retired swapchain that is not `oldSwapchain` -> `VK_ERROR_NATIVE_WINDOW_IN_USE_KHR`

Reply:
@ianelliottus I guess we are on a level playing field regarding to what I meant few months ago. 😄 Let's see...

So, we have
> `oldSwapchain` **must** not be in the retired state

and

> If `oldSwapchain` is not `VK_NULL_HANDLE`, `surface` **must** be associated with `oldSwapchain`

those are the undefined behaviors. Those two should be kinda equivalent to @critsec propositon, which says the same thing as one VU statement:

> `oldSwapchain` **must** be `VK_NULL_HANDLE` or a non-retired swapchain associated with `surface`.

(I like his proposition, because it does not silently assume `oldSwapchain` is valid unlike the first VU statement. I probably just forgot to update the PR.)

<hr/>

Now we need to cover the cases that are supposed to return `VK_ERROR_NATIVE_WINDOW_IN_USE_KHR`.  That is where @critsec is wrong. This pinned paragraph should describe only such cases (and VUs must not interfere and let those cases through, otherwise they would be undefined behavior). So it cannot be redundant to the VU (original, nor proposed).

>If `oldSwapchain` is `VK_NULL_HANDLE`, the swapchain **cannot** be created if the native window referred to by `surface` is already associated with a swapchain.

is supposed to describe the case nr. 2 in previous post.

>If `oldSwapchain` is a valid swapchain, the swapchain **cannot** be created if the native window referred to by `surface` is already associated with a non-retired swapchain not specified in `oldSwapchain`.

is supposed to describe the case nr. 5 in previous post.

<hr/>

Yes, the wording is bit confusing, but it should cover **all** the possible inputs (and their states).

Reply:
Oh yeah, I guess it actually conflicts with the VU.

We need to decide if nr. 5 should be undefined behavior or a `VK_ERROR_NATIVE_WINDOW_IN_USE_KHR`.

Technically it was undefided behavior (`must not`) before, then again `must not` was improperly used for the other cases too, so I cannot be 100 % sure this is what was initially meant.